### DISCUSSION TITLE
Uncaught Snapshot missing on Livewire component with id:

### DISCUSSION THREAD

**Discussion by nissicreative** [livewire/livewire#8921] [1. Help] [answered]
I have run into a frustrating issue after upgrading a project from Livewire 2 to Livewire 3. To make sure it was not due to something else in my codebase, I created a brand new Laravel project, installed Livewire, and set up a simplified scenario to reproduce the issue.

There is a `TaskGrid` component, which is basically an index page for a user’s tasks. The grid has one column (div) for new tasks and one column (div) for completed tasks. Each column contains the appropriate `TaskCard`s. So far, so good.

Each `TaskCard` has functionality that allows the task’s status to be updated (simplified in the sample code).

Calling the `updateStatus` action on the `TaskCard` component successfully updates the record in the database, but the UI does not respond appropriately. Instead, it throws a console error:

```
Uncaught Snapshot missing on Livewire component with id: 0egrJeCC6TxTlUZ2eEPh
```

If I inspect the DOM, I see an empty DIV where the component *should* now be rendered...

```html
<div wire:id="0egrJeCC6TxTlUZ2eEPh"></div>
```

I have tried every possible combination of `key` and `wire:key` I can think of and wrapped DIVs around everything I can think of, and nothing has solved the problem.

I am open to any suggestions!

Here is the code:

```php
<?php

namespace App\Livewire;

use App\Models\Task;
use Livewire\Attributes\On;
use Livewire\Component;

class TaskGrid extends Component
{
    #[On('task-status-updated')]
    public function render()
    {
        return view('livewire.task-grid', [
            'newTasks' => Task::where('status', 'new')->get(),
            'completedTasks' => Task::where('status', 'completed')->get(),
        ]);
    }
}
```

`task-grid.blade.php`
```blade
<div class="grid grid-cols-2 gap-6">
  <div class="space-y-6"
    id="new-tasks">
    <h2>New Tasks</h2>

    @foreach ($newTasks as $task)
      <livewire:task-card :task="$task"
        :key="$task->id" />
    @endforeach
  </div>

  <div class="space-y-6"
    id="completed-tasks">
    <h2>Completed Tasks</h2>

    @foreach ($completedTasks as $task)
      <livewire:task-card :task="$task"
        :key="$task->id" />
    @endforeach
  </div>
</div>

```

```php
<?php

namespace App\Livewire;

use App\Models\Task;
use Livewire\Component;

class TaskCard extends Component
{
    public Task $task;

    public function updateStatus($status)
    {
        $this->task->update([
            'status' => $status,
        ]);

        $this->dispatch('task-status-updated');
    }

    public function render()
    {
        return view('livewire.task-card');
    }
}
```

`task-card.blade.php`
```blade
<div class="border p-4">
  <div>{{ $task->id }}: {{ $task->name }}</div>

  <div class="grid grid-cols-2 gap-6">
    <button type="button"
      wire:click="updateStatus('new')">New</button>
    <button type="button"
      wire:click="updateStatus('completed')">Completed</button>
  </div>
</div>
```

Thanks,
Mike

**Comment by nissicreative** [comment, marked as answer]
In case anyone else stumbles across this in the future, I found a solution (workaround?) by appending `time()` to the `key`, forcing the keys to regenerate when the parent re-renders. (See also https://github.com/livewire/livewire/discussions/8575)

task-grid.blade.php
```blade
<div class="grid grid-cols-2 gap-6">
  <div class="space-y-6"
    id="new-tasks">
    <h2>New Tasks</h2>

    @foreach ($newTasks as $task)
      <livewire:task-card :task="$task"
        :key="$task->id . time()" />
    @endforeach
  </div>

  <div class="space-y-6"
    id="completed-tasks">
    <h2>Completed Tasks</h2>

    @foreach ($completedTasks as $task)
      <livewire:task-card :task="$task"
        :key="$task->id . time()" />
    @endforeach
  </div>
</div>
```

**Reply by ojessecruz** [reply]
Thanks for coming back and write what solves your problem. I am facing the same issue, but I can't reproduce this. It also started after upgrade to Livewire 3.

I am struggle with this for some time already, adding keys for each loop and also rebuilding from the ground some components. 
Let's hope this solution works for me as well.

**Reply by geechartier** [reply]
I have just upgraded my site to Livewire 3, and had the exact same problem. Adding `. time()` stopped me getting the error but caused other issues because it doesn't resolve the problem that the original snapshot was lost, along with its state/properties. Fortunately for me, I have two fairly similar Livewire components that both make use of other Livewire components. One of mine was working but the other was not. By a process of elimination, I found that the snapshot missing error was caused by one of my Livewire components having a property called `id`. When I removed that, the snapshot error went away.

**Reply by geechartier** [reply]
Although removing the `id` property on my component fixed the problem for me, I am now a little confused as to why, given that [this example](https://livewire.laravel.com/docs/properties#dont-trust-property-values) on the Livewire site uses a Livewire component with an `id` property.

**Reply by bigboypantson** [reply]
Strangely enough, I only get this problem on production env.

**Reply by bigboypantson** [reply]
I managed to sort this on my project, by moving a livewire component up a few levels in my templates so it wasn't nested inside the other livewire component, really odd as worked fine on local and prod is the same config, same setup etc.

**Reply by hasan-aftab** [reply]
@nissicreative thanks for saving life

**Comment by joshhanley** [comment]
We've just done a bunch of work to try and fix scenarios that cause this error. The fixes will be in the next release.

**Reply by patrickweh** [reply]
Hey Josh,

Thanks for fixing this :)

Do you have an idea when the next release will happen?

**Reply by joshhanley** [reply]
@patrickweh all things going well, it should be this week, but no guarantees 🙂

**Comment by binkapS** [comment]
This happens each time i cache the views and livewire send a request
If i clear the cached views no problems

i provided keys for child components
